### PR TITLE
Update hadoop to 3.2.0

### PR DESCRIPTION
**Pull request by rajatrj20**
When presto runs on Hadoop 3 with `dfs.encrypt.data.transfer=true`, queries fails with error:

> com.facebook.presto.spi.PrestoException: Error committing write to Hive 
> 
> Caused by: org.apache.hadoop.ipc.RemoteException: File /tmp/presto-hadoop/a38381f4-fcc3-431d-bd73-bb0d872a1572/20211106_073946_00003_k9xuf_070809d7-6c9b-4d5f-ac17-86a998d2958d could only be written to 0 of the 1 minReplication nodes. There are 3 datanode(s) running and 3 node(s) are excluded in this operation.

and on HDFS side following error is thrown:
>javax.security.sasl.SaslException: Invalid token in javax.security.sasl.qop: 
    at com.sun.security.sasl.util.AbstractSaslImpl.parseProp(AbstractSaslImpl.java:242)
    at com.sun.security.sasl.util.AbstractSaslImpl.parseQop(AbstractSaslImpl.java:206)
    at com.sun.security.sasl.util.AbstractSaslImpl.parseQop(AbstractSaslImpl.java:197)
    at com.sun.security.sasl.util.AbstractSaslImpl.<init>(AbstractSaslImpl.java:73)
    at com.sun.security.sasl.digest.DigestMD5Base.<init>(DigestMD5Base.java:174)
    at com.sun.security.sasl.digest.DigestMD5Server.<init>(DigestMD5Server.java:145)
    at com.sun.security.sasl.digest.FactoryImpl.createSaslServer(FactoryImpl.java:109)
    at javax.security.sasl.Sasl.createSaslServer(Sasl.java:567)
    at org.apache.hadoop.hdfs.protocol.datatransfer.sasl.SaslParticipant.createServerSaslParticipant(SaslParticipant.java:66)
    at org.apache.hadoop.hdfs.protocol.datatransfer.sasl.SaslDataTransferServer.doSaslHandshake(SaslDataTransferServer.java:385)
    at org.apache.hadoop.hdfs.protocol.datatransfer.sasl.SaslDataTransferServer.getEncryptedStreams(SaslDataTransferServer.java:186)
    at org.apache.hadoop.hdfs.protocol.datatransfer.sasl.SaslDataTransferServer.receive(SaslDataTransferServer.java:118)
    at org.apache.hadoop.hdfs.server.datanode.DataXceiver.run(DataXceiver.java:234)
    at java.lang.Thread.run(Thread.java:748)

This happens because presto still uses hadoop 2 libraries which results in failure when running on hadoop 3 system. This PR is to update support of hadoop 3 to presto engine by upgrading hadoop to version 3.2.0. This changes are not tested on hadoop 2 environment after updating it to hadoop 3 and if visibility is needed I can update that.

To add some historical context on how PrestoDB moved from hadoop-apache1 to hadoop-apache2, Presto was using hadoop-apache1 and then plugin for hadoop-apache2 was [added](https://github.com/prestodb/presto/pull/943) as well. After sometime plugin for hadoop-apache1 was [removed](https://github.com/prestodb/presto/pull/8213) from the dependencies and it started using only hadoop-apache2.
Also, Trino has moved away from hadoop 2 and is now on hadoop 3. Reference: [Update to Hadoop 3.2.0 #3](https://github.com/trinodb/trino-hadoop-apache/pull/3)

Ideally, we should also look at renaming this repository to **_presto-hadoop-apache_** to make it more generic or a new repository **_presto-hadoop-apache3_** should be created for hadoop version 3 the way repo [**_presto-hadoop-apache1_**](https://github.com/prestodb/presto-hadoop-apache1) and [**_presto-hadoop-apache2_** ](https://github.com/prestodb/presto-hadoop-apache2)already exist for version 1 and 2 respectively. Since, I am not sure which route needs to be taken, I didn't create or tried to rename the repository. Please let me know on the steps to be taken and I can follow those steps to do the further work required.

### CODE REVIEW

**Comment by rajatrj20** [comment]
@aweisberg @imjalpreet Please help reviewing this PR. I have opened this PR based on the feedback and comment from #53

**Comment by rajatrj20** [comment]
@imjalpreet, Could you please go through this CR?

**Comment by imjalpreet** [comment]
@rajatrj20 Sorry for the delay, this PR completely got out of my mind. I will review it this week.

**Comment by imjalpreet** [comment]
@rajatrj20 I will try to submit my review this week, got caught up in some other work for the past few weeks. Just carrying out some tests and after that will add my review.

**Review by imjalpreet — CHANGES_REQUESTED** [review]
@rajatrj20 Thanks for the contribution. Sorry for the long delay in reviewing this PR, I will try to fast-track it now. I have done a first pass and had a few comments. I will add more comments in case I have one. 

Also, have you carried out tests by including this updated dependency with Presto in your environment?

We would also need to plan on how we can introduce product/integration tests with Hadoop 3.x as well. Right now all the docker images used in the product tests are Hadoop 2.7.x.

**Comment by rohanpednekar** [comment]
@rajatrj20 Thanks for your contributions, do you think you can address the feedback, so we can target this feature for the next release?

**Comment by rajatrj20** [comment]
@imjalpreet thanks for the feedback. Will work on this and update.

File: src/main/resources/core-site.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0"?>
```
[the remark this answers is not in the document]
Comment (rajatrj20) [reply src/main/resources/core-site.xml]: Yes, this is no longer required and I have removed it.

**Comment by rajatrj20** [comment]
> @rajatrj20 Thanks for the contribution. Sorry for the long delay in reviewing this PR, I will try to fast-track it now. I have done a first pass and had a few comments. I will add more comments in case I have one.
> 
> Also, have you carried out tests by including this updated dependency with Presto in your environment?
> 
> We would also need to plan on how we can introduce product/integration tests with Hadoop 3.x as well. Right now all the docker images used in the product tests are Hadoop 2.7.x.

I have run basic product tests by including this updated dependency and they are passing.

Yes, we will need to docker images to use updated version of Hadoop and other dependencies after this upgrade.

**Comment by rajatrj20** [comment]
@imjalpreet, just following up on the review. Please help review updated changes.

**Review by imjalpreet — CHANGES_REQUESTED** [review]
@rajatrj20 thanks for the changes, a few more minor comments, let me know if you have any questions.

File: src/main/resources/nativelib/Linux-amd64/libhadoop.so
[the remark this answers is not in the document]
Comment (rajatrj20) [reply src/main/resources/nativelib/Linux-amd64/libhadoop.so:1]: No, I have taken them from the PR: https://github.com/trinodb/trino-hadoop-apache/pull/3

File: src/main/resources/nativelib/Linux-amd64/libzstd.so
Comment (imjalpreet) [inline src/main/resources/nativelib/Linux-amd64/libzstd.so:1]: Forgot to mention, let's keep the permissions as 755 itself which were set earlier.

Comment (rajatrj20) [reply src/main/resources/nativelib/Linux-amd64/libzstd.so:1]: Removed checkstyle changes and changed permission to the previous state.

**Comment by rajatrj20** [comment]
@imjalpreet can you please review the changes.

**Review by imjalpreet — APPROVED** [review]
LGTM 👍🏼

**Comment by rajatrj20** [comment]
> LGTM 👍🏼

Thanks for reviewing. I don't have permission to merge this. Could you please merge this?

**Comment by rajatrj20** [comment]
@imjalpreet Can we please get this merged?

**Comment by imjalpreet** [comment]
@tdcmeehan can you please do a final review? If you believe there is someone else who should also do a round of review, please add. Thanks.

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**